### PR TITLE
Improve tx handling; remove web3-core dependency

### PR DESCRIPTION
**Pull request by rmeissner**
- Improve tx handling: If transaction was submitted via the provider, we will cache a fake tx to be returned in case we cannot load the real one.
- Remove web3-core dependency (TODO: cleanup yarn.lock after each package release)

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11956,7 +11956,7 @@ web3-core-subscriptions@1.3.4:
     underscore "1.9.1"
     web3-core-helpers "1.3.4"
 
-web3-core@1.3.4, web3-core@^1.3.0:
+web3-core@1.3.4:
```
Comment (mikheevm) [inline yarn.lock:11959]: it's still installed 😛

File: packages/safe-apps-sdk/package.json [hunk 2]
```diff
@@ -37,9 +36,7 @@
     "prettier": "^2.2.1",
     "rimraf": "^3.0.2"
   },
-  "peerDependencies": {
-    "web3-core": "^1.3.0"
-  },
+  "peerDependencies": {},
```
Comment (mikheevm) [inline packages/safe-apps-sdk/package.json:39]: you can exclude "peerDependencies" at all

File: packages/safe-apps-provider/src/provider.ts [hunk 3]
```diff
@@ -92,7 +106,12 @@ export class SafeAppProvider implements AsyncSendable {
           const resp = await this.sdk.txs.getBySafeTxHash(txHash);
           txHash = resp.transactionHash || txHash;
         } catch (e) {}
-        return this.sdk.eth.getTransactionByHash([txHash]).then((tx) => {
+        // Use fake transaction if we don't have a real tx hash
+        if (this.submittedTxs.has(txHash)) {
+          return this.submittedTxs.get(txHash);
+        }
+        // eslint-disable-next-line
+        return this.sdk.eth.getTransactionByHash([txHash]).then((tx: any) => {
```
Comment (mikheevm) [inline packages/safe-apps-provider/src/provider.ts:114]: why any? the SDK should have a correct return type

File: packages/safe-apps-provider/src/provider.ts [hunk 4]
```diff
@@ -106,7 +125,8 @@ export class SafeAppProvider implements AsyncSendable {
           const resp = await this.sdk.txs.getBySafeTxHash(txHash);
           txHash = resp.transactionHash || txHash;
         } catch (e) {}
-        return this.sdk.eth.getTransactionReceipt([txHash]).then((tx) => {
+        // eslint-disable-next-line
+        return this.sdk.eth.getTransactionReceipt([txHash]).then((tx: any) => {
```
Comment (mikheevm) [inline packages/safe-apps-provider/src/provider.ts:129]: same here

Comment (rmeissner) [reply packages/safe-apps-provider/src/provider.ts:129]: it was like that before ... not sure why it did not complain

File: packages/safe-apps-sdk/package.json (hunk 2 above)
Comment (rmeissner) [reply packages/safe-apps-sdk/package.json:39]: I used `yarn remove` and that one didn't remove it, so I didn't touch it further

File: yarn.lock (hunk 1 above)
Comment (rmeissner) [reply yarn.lock:11959]: yeah, as it some of the packages still have the older version installed and I can only completely update that once everything is on npm

File: packages/safe-apps-provider/src/provider.ts (hunk 4 above)
Comment (mikheevm) [reply packages/safe-apps-provider/src/provider.ts]: Screenshot 2021-03-29 at 13 08 53

for me the type is inherited and it should be like that

Comment (rmeissner) [reply packages/safe-apps-provider/src/provider.ts]: interesting 🤔  ... anyways I set it now ... so it is correct in any way

Comment (rmeissner) [reply packages/safe-apps-provider/src/provider.ts]: actually it does for me too now ... ok I will remove the import ... not sure what vscode did there

**Review by mikheevm — APPROVED** [review]

**Pull request merged**